ArmTemplates/app-gateway-v2.md

```diff
@@ -0,0 +1,239 @@
+# Application Gateway v2
+
+Adds an Application Gateway v2
+
+## parameters
+
+appGatewayName: (required) string
+
+Name of the application gateway resource to create
+
+subnetRef: (required) string
+
+The subnet resource ID where the application gateway will go.
+
+If you have the vnet and subnet names, you can get the ID with the following function
+`resourceId('Microsoft.Network/virtualNetworks/subnets', 'vnet', 'subnet')`
+
+appGatewayTier: (optional) string
+
+Application gateway type and instance size combined
+Must be either Standard_v2 or WAF_v2
+
+backendPools: (required) array of object
+
+A list of backend pools to create.
+The first backend pool specified will be the default one (used if not routing)
+
+Each backend is specified by an object consisting of
+
+* name: the name the backend pool resource
+* fqdn: the full domain name
+
+An example of a valid object
+
+```json
+{
+    "name": "backendName",
+    "fqdn": "be.example.net"
+}
+```
+
+backendHttpSettings: (required) array of object
+
+A list of settings for connecting to the back end application(s).
+The first setting in the array is used for the default routing rule.
+
+Each backend setting is specified by an object consisting of
+
+* name: the name the backend setting
+* port: port
+* protocol: protocol
+* hostnameFromBackendAddress: Select hostname from backend address
+* timeout: set the request timeout (optional)
+* backendPath: override the backend path (optional)
+* probeName: name of probe (optional)
+* authCerts: array of authentication certificates (optional)

```

Comment:
What format are the certificates supplied in? (ie KeyVault url)

Reply:
They are IDs to existing certs in Azure. I've added a link to the Microsoft doc for further information but I've not used this option so I've not got a tested example to include.